### DISCUSSION TITLE
Como saber se alguém já é um bom programador?

### DISCUSSION THREAD

**Discussion by pauloviniciusdev** [frontendbr/forum#2119] [Perguntas e Respostas]
Olá! sou um iniciante no mundo da programação e queria saber como avaliar o meu nível em linguagens como, por exemplo, o JavaScript ou qualquer outra linguagem. Existem sites que dão uma nota para nós ou algo do gênero?

**Comment by rapharma** [comment]
É uma pergunta muito abrangente, mas tem o w3schools, o linkedin e pluralsight que eu conheço que possuem testes

**Reply by pauloviniciusdev** [reply]
Somente isso já foi de grande ajuda, amigo. Obrigado!

**Comment by NayRSilva** [comment]
Olha, fora testes, um jeito mais qualitativo de avaliar sua habilidade em uma linguagem seria fazer projetos dela. Porque os testes vao te perguntar muitas coisas, alguns perguntam sobre funcionalidades que quase ninguem usa, outros sao mais voltados pra algoritmos ou coisas mais teoricas. 

Conforme voce evolui, você começa a perceber quais projetos você tem confiança de levar e quais não. Programação é algo muito amplo, e depende muito do seu proprio caminho como profissional e seus objetivos. Conheço muito front-ender bacana que chora quando tem que fazer efeito simples de transição, e vejo muita gente com projetos incriveis no codepen que nao tanka um site.

Então assim, respondendo sua pergunta geral, se você quer saber como é que ta seu nivel como programador, escolhe primeiro o que voce tem interesse de fazer, procura algum curso gratis com projetos e descobre até onde voce consegue levar. Por exemplo, se a gente fala de front-end, existe no freecodecamp uma trilha pra voce aprender projetos web e responsividade. Voce pode verificar os projetos finais dessa trilha e no mínimo tentar cria-los. Acho que isso da uma ideia boa da capacidade, e voce sempre pode comparar seus projetos novos com antigos e ver como evoluiu.

Se voce quer algo mais como um hank, ou saber como ta seu nivel de algoritmo um site bom pra isso é o hackerrank (acho que é esse o nome). Existe um similar em portugues mas eu nao vou saber dizer qual é. Ele te da simples tarefas de programação e conforme voce vai passando ele vai aumentando o nivel. E ele tambem rankeia as melhores respostas

Mas essa real nao é uma pergunta trivial, tanto é que varias empresas que procuram programadores tem diversos criterios de escolha, do curriculo ao portifolio e testes de algoritmos

**Reply by pauloviniciusdev** [reply]
Muito obrigado, Nayara. Sua resposta acabou com minha dúvida e coisas que eu também nem sabia.

**Comment by ninetails** [comment]
Também recomendo o Hackerrank caso quiser se testar.

Mas na minha opinião, creio que o maior ponto de virada que diz que vc sabe a linguagem, framework, conhecimento e tal é na hora que você parar de acompanhar os artigos de "Como fazer X ou Y" e começa procurar "Quais as mudanças na próxima versão"...

Mas isso é sobre domínio da linguagem ou do conhecimento que se está testando.

Já pra pergunta do título, que acho bem diferente do que foi perguntado no corpo da mensagem, ser um bom programador é bem subjetivo. Mesmo porquê a senioridade pode variar de empresa a empresa pois depende do ramo de atuação, stack utilizada, domínio sobre o produto etc. Mas creio que subjetivamente pra mim, ser um bom programador na média é quando vc, não que para de aprender pois isso nunca paramos, mas quando começa a ensinar outros e ser referência para o tal. Também que saiba boas práticas, escrever código legível, desacoplado, testado, escalável e possivelmente reutilizável... Para outros pode ser que ser um bom programador é ser aquele que sempre entrega o que lhe foi delegado sem muitos atrasos.... por aí vai. É algo que só pode ser avaliado por um contexto, seja o programador dentro de uma empresa, ou dentro de uma comunidade, gerando conteúdos...

(bem filosófico rs)

**Reply by pauloviniciusdev** [reply]
Foi mal ai, meu brother! Da próxima eu penso melhor antes de formular uma pergunta. 😅
Obrigado ai pela resposta.

**Comment by nataliapastre-dev** [comment]
Não acho que exista uma nota que consiga definir se alguém é um bom programador. Na minha opinião, a melhor forma de avaliar a evolução é observar se você consegue resolver problemas cada vez mais complexos, entender o código que está escrevendo e construir projetos sem depender sempre de tutoriais. Plataformas como GitHub e desafios de programação podem ajudar, mas acredito que projetos próprios são uma ótima forma de perceber essa evolução. Também é importante lembrar que programação envolve aprendizado constante, então mesmo desenvolvedores experientes continuam estudando.

**Reply by Yures1lva** [reply]
Quando se já um bom, em meia hora de conversa já da pra saber, apenas avaliar como a pessoa fala sobre certo assuntos técnicos  pelo menos na minha visão acredito que seja uma questão mais de carater e capacidade da pessoa do que qualidade técnica alinhada com anos de xp.
